examples/demo_threading/CMakeLists.txt

```diff
@@ -46,14 +46,14 @@ if(NOT CxxQt_FOUND)
     FetchContent_Declare(
         CxxQt
         GIT_REPOSITORY https://github.com/kdab/cxx-qt-cmake.git
-        GIT_TAG main
+        GIT_TAG cmake-qt-modules
     )
 
     FetchContent_MakeAvailable(CxxQt)
 endif()
 
 set(CRATE cxx_qt_demo_threading)
-cxx_qt_import_crate(MANIFEST_PATH rust/Cargo.toml CRATES ${CRATE})
+cxx_qt_import_crate(MANIFEST_PATH rust/Cargo.toml CRATES ${CRATE} QT_MODULES Qt::Core Qt::Gui Qt::Qml Qt::QuickControls2)
```

Comment:
Can we now maybe remove some of the dependencies from the actual app?
E.g. QuickControls2 probably is only needed by our own QML code, right?
Not by main.cpp

Reply:
In theory, although depends if we want to be rebuilding cxx-qt-lib for every example (as is happening now) or just enable qt_full + full everywhere. I'll see if i can match some of them.